block-mode is not compatible with aes ( the following trait bounds were not satisfied:  aes::Aes128: block_cipher_trait::BlockCipher)

The latest published version of block-mode does not seem to work with the latest published version of aes. The first example in the documentation does not compile.

### Cargo.toml

```toml
[package]
name = "test-aes-cbc"
version = "0.1.0"
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
block-modes = "0.3"
aes-soft = "0.4"
hex-literal = "0.2"
```

### main.rs

(taken from https://docs.rs/block-modes/latest/block_modes/index.html)

```rust
#[macro_use] extern crate hex_literal;
extern crate aes_soft as aes;
extern crate block_modes;

use block_modes::{BlockMode, Cbc};
use block_modes::block_padding::Pkcs7;
use aes::Aes128;

// create an alias for convinience
type Aes128Cbc = Cbc<Aes128, Pkcs7>;

fn main() {
    let key = hex!("000102030405060708090a0b0c0d0e0f");
    let iv = hex!("f0f1f2f3f4f5f6f7f8f9fafbfcfdfeff");
    let plaintext = b"Hello world!";
    let cipher = Aes128Cbc::new_var(&key, &iv).unwrap();

    // buffer must have enough space for message+padding
    let mut buffer = [0u8; 32];
    // copy message to the buffer
    let pos = plaintext.len();
    buffer[..pos].copy_from_slice(plaintext);
    let ciphertext = cipher.encrypt(&mut buffer, pos).unwrap();

    assert_eq!(ciphertext, hex!("1b7a4c403124ae2fb52bedc534d82fa8"));

    // re-create cipher mode instance
    let cipher = Aes128Cbc::new_var(&key, &iv).unwrap();
    let mut buf = ciphertext.to_vec();
    let decrypted_ciphertext = cipher.decrypt(&mut buf).unwrap();

    assert_eq!(decrypted_ciphertext, plaintext);
}
```

## Compilation result

```
   Compiling test-aes-cbc v0.1.0 (/tmp/test-aes-cbc)
error[E0599]: no function or associated item named `new_var` found for struct `block_modes::Cbc<aes::Aes128, block_modes::block_padding::Pkcs7>` in the current scope
  --> src/main.rs:16:29
   |
16 |     let cipher = Aes128Cbc::new_var(&key, &iv).unwrap();
   |                             ^^^^^^^ function or associated item not found in `block_modes::Cbc<aes::Aes128, block_modes::block_padding::Pkcs7>`
   | 
  ::: /home/ophir/.cargo/registry/src/github.com-1ecc6299db9ec823/block-modes-0.3.3/src/cbc.rs:12:1
   |
12 | pub struct Cbc<C: BlockCipher, P: Padding> {
   | ------------------------------------------ doesn't satisfy `_: block_modes::BlockMode<aes::Aes128, block_modes::block_padding::Pkcs7>`
   | 
  ::: /home/ophir/.cargo/registry/src/github.com-1ecc6299db9ec823/aes-soft-0.4.0/src/impls.rs:28:9
   |
28 |         pub struct $name {
   |         ---------------- doesn't satisfy `aes::Aes128: block_cipher_trait::BlockCipher`
   |
   = note: the method `new_var` exists but the following trait bounds were not satisfied:
           `aes::Aes128: block_cipher_trait::BlockCipher`
           which is required by `block_modes::Cbc<aes::Aes128, block_modes::block_padding::Pkcs7>: block_modes::BlockMode<aes::Aes128, block_modes::block_padding::Pkcs7>`

error[E0277]: the trait bound `aes::Aes128: block_cipher_trait::BlockCipher` is not satisfied
  --> src/main.rs:16:18
   |
16 |     let cipher = Aes128Cbc::new_var(&key, &iv).unwrap();
   |                  ^^^^^^^^^^^^^^^^^^ the trait `block_cipher_trait::BlockCipher` is not implemented for `aes::Aes128`
   | 
  ::: /home/ophir/.cargo/registry/src/github.com-1ecc6299db9ec823/block-modes-0.3.3/src/cbc.rs:12:19
   |
12 | pub struct Cbc<C: BlockCipher, P: Padding> {
   |                   ----------- required by this bound in `block_modes::Cbc`

error[E0599]: no function or associated item named `new_var` found for struct `block_modes::Cbc<aes::Aes128, block_modes::block_padding::Pkcs7>` in the current scope
  --> src/main.rs:28:29
   |
28 |     let cipher = Aes128Cbc::new_var(&key, &iv).unwrap();
   |                             ^^^^^^^ function or associated item not found in `block_modes::Cbc<aes::Aes128, block_modes::block_padding::Pkcs7>`
   | 
  ::: /home/ophir/.cargo/registry/src/github.com-1ecc6299db9ec823/block-modes-0.3.3/src/cbc.rs:12:1
   |
12 | pub struct Cbc<C: BlockCipher, P: Padding> {
   | ------------------------------------------ doesn't satisfy `_: block_modes::BlockMode<aes::Aes128, block_modes::block_padding::Pkcs7>`
   | 
  ::: /home/ophir/.cargo/registry/src/github.com-1ecc6299db9ec823/aes-soft-0.4.0/src/impls.rs:28:9
   |
28 |         pub struct $name {
   |         ---------------- doesn't satisfy `aes::Aes128: block_cipher_trait::BlockCipher`
   |
   = note: the method `new_var` exists but the following trait bounds were not satisfied:
           `aes::Aes128: block_cipher_trait::BlockCipher`
           which is required by `block_modes::Cbc<aes::Aes128, block_modes::block_padding::Pkcs7>: block_modes::BlockMode<aes::Aes128, block_modes::block_padding::Pkcs7>`

error[E0277]: the trait bound `aes::Aes128: block_cipher_trait::BlockCipher` is not satisfied
  --> src/main.rs:28:18
   |
28 |     let cipher = Aes128Cbc::new_var(&key, &iv).unwrap();
   |                  ^^^^^^^^^^^^^^^^^^ the trait `block_cipher_trait::BlockCipher` is not implemented for `aes::Aes128`
   | 
  ::: /home/ophir/.cargo/registry/src/github.com-1ecc6299db9ec823/block-modes-0.3.3/src/cbc.rs:12:19
   |
12 | pub struct Cbc<C: BlockCipher, P: Padding> {
   |                   ----------- required by this bound in `block_modes::Cbc`

warning: unused import: `BlockMode`
 --> src/main.rs:5:19
  |
5 | use block_modes::{BlockMode, Cbc};
  |                   ^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error: aborting due to 4 previous errors; 1 warning emitted

Some errors have detailed explanations: E0277, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `test-aes-cbc`.

To learn more, run the command again with --verbose.
```

--- a/block-modes/CHANGELOG.md
+++ b/block-modes/CHANGELOG.md
@@ -5,6 +5,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 0.4.0 (2020-06-07)
+### Changed
+- Upgrade to Rust 2018 edition ([#87])
+- Bump `block-cipher` dependency to v0.7 ([#87])
+
+[#87]: https://github.com/RustCrypto/block-ciphers/pull/87
+
 ## 0.3.3 (2019-04-28)
 
 ## 0.3.2 (2019-02-04)

--- a/block-modes/Cargo.toml
+++ b/block-modes/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "block-modes"
-version = "0.4.0-pre"
+version = "0.4.0"
 description = "Block cipher modes of operation"
 authors = ["RustCrypto Developers"]
 license = "MIT OR Apache-2.0"

